code/centauri/algorithms/grandpa/verifier/src/lib.rs

```diff
@@ -69,18 +69,19 @@ where
 		Err(anyhow!("Latest finalized block should be highest block in unknown_headers"))?;
 	}
 
-	let from = finality_proof
-		.unknown_headers
-		.iter()
-		.min_by_key(|h| *h.number())
-		.ok_or_else(|| anyhow!("Unknown headers can't be empty!"))?;
-	let mut finalized = headers
-		.ancestry(from.hash(), target.hash())
-		.map_err(|_| anyhow!("Invalid ancestry!"))?;
+	let from = client_state.latest_relay_hash;
```

Comment:
seeing as there might be relay chain headers < `client_state.latest_relay_hash`, we should also check that there exists a route between the minimum header in unknown headers to `client_state.latest_relay_hash`.